### PR TITLE
hclfmt: init at 2.12.0

### PR DESCRIPTION
**Pull request by zimbatm**
###### Description of changes

new package!

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/hclfmt/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+buildGoModule rec {
```
Comment (winterqt) [inline pkgs/development/tools/hclfmt/default.nix:2]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:

buildGoModule (finalAttrs: {
```

File: pkgs/development/tools/hclfmt/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  subPackages = [ "cmd/hclfmt" ];
+
+  meta = {
+    description = "a code formatter for the Hashicorp Configuration Language (HCL) format";
+    homepage = "https://github.com/hashicorp/hcl/tree/main/cmd/hclfmt";
+    license = lib.licenses.mpl20;
+    maintainers = [
+      lib.maintainers.zimbatm
+    ];
+  };
```
Comment (winterqt) [inline pkgs/development/tools/hclfmt/default.nix:26]:
```suggestion
  meta = with lib; {
    description = "a code formatter for the Hashicorp Configuration Language (HCL) format";
    homepage = "https://github.com/hashicorp/hcl/tree/main/cmd/hclfmt";
    license = licenses.mpl20;
    maintainers = with maintainers; [ zimbatm ];
  };
```

File: pkgs/development/tools/hclfmt/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+buildGoModule rec {
+  pname = "hclfmt";
+  version = "2.12.0";
+
+  src = fetchFromGitHub {
+    owner = "hashicorp";
+    repo = "hcl";
+    rev = "v${version}";
```
Comment (winterqt) [inline pkgs/development/tools/hclfmt/default.nix:9]:
```suggestion
    rev = "v${finalAttrs.version}";
```

File: pkgs/development/tools/hclfmt/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = {
+    description = "a code formatter for the Hashicorp Configuration Language (HCL) format";
+    homepage = "https://github.com/hashicorp/hcl/tree/main/cmd/hclfmt";
+    license = lib.licenses.mpl20;
+    maintainers = [
+      lib.maintainers.zimbatm
+    ];
+  };
+}
```
Comment (winterqt) [inline pkgs/development/tools/hclfmt/default.nix:27]:
```suggestion
})
```

**Review by winterqt — CHANGES_REQUESTED** [review]
Just some minor formatting changes. Looks good otherwise, though!

**Comment by zowoq** [comment]
Would have be good to either fix the darwin builds or mark them as broken ...

**Pull request merged**